warehouse/macaroons/services.py

```diff
@@ -135,7 +135,11 @@ def create_macaroon(self, location, user_id, description, caveats):
         """
         user = self.db.query(User).filter(User.id == user_id).one()
 
-        dm = Macaroon(user=user, description=description, caveats=caveats)
+        # NOTE: This is a bit of a hack: we keep a separate copy of the
+        # permissions caveat in the DB, so that we can display scope information
+        # in the UI.
+        permissions = next(c["permissions"] for c in caveats if "permissions" in c)
+        dm = Macaroon(user=user, description=description, permissions=permissions)
```

Comment:
Highlighting this as a bit of a hack: I'm not really happy with this trick, but the alternative would be a `create_macaroon` signature like this:

```python
def create_macaroon(self, location, user_id, description, caveats, permissions):
  ...
```

...where `caveats[N] == permissions`. That feels roughly as bad, so I figured reducing the parameter deduplication was a reasonable tradeoff. But I'm very open to better ideas.